src/client/utils/tracking/hooks/useTrackPageViewEvent.ts

```diff
@@ -13,6 +13,8 @@ export const useTrackPageViewEvent = () => {
   const market = useMarket().toLowerCase()
   const location = useLocation()
 
+  const [currentPathname, setCurrentPathname] = useState<string>()
```

Comment:
```suggestion
  const [currentPathname, setCurrentPathname] = useState<string>(location.pathname)
```

...but I'm not sure I like this. We should avoid state unless we really care about it. This is a bit murky. Isn't it possible to subscribe to some event that fires every time there's a page navigation? Like:

```
history.listen((location, action) => {
    // location is an object like window.location
    console.log(action, location.pathname, location.state)
});
```

Reply:
problem is location object can trigger changes with location.search as well, but we're interested in pathname changes only, so we need a way to keep track of the previous pathname

Reply:
@robinandeer I reverted this change so this fix will come in a future PR where we can discuss the solution in more details